### PR TITLE
Adding doc on - How to Use W3C Trace Headers in Dapr

### LINKED ISSUE #564: Create howto docs to use W3C trace headers
## Describe the proposal

* [ ] Create the doc to use W3C trace headers from api to app channel
* [ ] Create the doc for how to correlate with user app tracing and dapr sidecar tracing

### PR DESCRIPTION
**Pull request by shalabhms**
Adding How to use W3C trace context in Dapr calls.

## Issue reference

Closes #564

### CODE REVIEW

File: howto/use-w3c-tracecontext/README.md [hunk 1]
```diff
@@ -0,0 +1,243 @@
+# W3C Trace Context - Background
```
Comment (orizohar) [inline howto/use-w3c-tracecontext/README.md:1]: Seems like "Background" should be the header for the first section not the entire article

File: howto/use-w3c-tracecontext/README.md [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+| Language | SDK |
+|:-------:|:----:|
+| Go | [Link](https://pkg.go.dev/go.opencensus.io?tab=overview)
+| Java | [Link](https://www.javadoc.io/doc/io.opencensus/opencensus-api/latest/index.html)
+| C# | [Link](https://github.com/census-instrumentation/opencensus-csharp/)
+| C++ | [Link](https://github.com/census-instrumentation/opencensus-cpp)
+| Node.js | [Link](https://github.com/census-instrumentation/opencensus-node)
+| Python | [Link](https://census-instrumentation.github.io/opencensus-python/trace/api/index.html)
+
+For the purpose of example, we will use the same [grpc app](../create-grpc-app) based on Go SDK.
```
Comment (orizohar) [inline howto/use-w3c-tracecontext/README.md:74]: Please rephrase to remove all first person references ("we", "I")

File: howto/use-w3c-tracecontext/README.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+`$ go get -u go.opencensus.io`
+
+1. import the package "go.opencensus.io/trace"
+2. Create trace context
+
+When you want to pass the trace context , you will be starting the trace spans. At high level, since a distributed trace tracks the progression of a single user request as it is handled by the services and processes that make up an application. Each step is called a span in the trace. Spans include metadata about the step, including especially the time spent in the step, called the span’s latency.
+
+So Span is the unit step in a trace. Each span has a name, latency, status and additional metadata.
+
+Below we are starting a span for a cache read and ending it when we are done:
```
Comment (orizohar) [inline howto/use-w3c-tracecontext/README.md:89]: Please rephrase to remove all first person references ("we", "I")

File: howto/use-w3c-tracecontext/README.md [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+```
+
+The StartSpan call returns 2 values. If you want to propagate trace context within your calls in the same process, you can use `context.Context` to propagate spans.
+The returned `span` has the fields 'TraceID' and 'SpanID'. 
+
+You can read more on these fields usage and details [here](https://opencensus.io/tracing/span/)
+
+When you call the Dapr API through HTTP/gRPC, you need to pass the trace context across process. For across the network, OpenCensus provides different propagation methods for different protocols. You can read about the propagation package [here](https://godoc.org/go.opencensus.io/trace/propagation).
+
+In our example, we will use gRPC calls , for that OpenCensus’ binary propagation format is used.
```
Comment (orizohar) [inline howto/use-w3c-tracecontext/README.md:105]: Please rephrase to remove all first person references ("we", "I")

File: howto/use-w3c-tracecontext/README.md [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+req, _ := http.NewRequest("GET", "http://localhost:3500/v1.0/invoke/mathService/method/api/v1/add", nil)
+
+traceContext := span.SpanContext()
+f.SpanContextToRequest(traceContext, req)
+```
+
+Please go here for [Dapr API reference](https://github.com/dapr/docs/tree/master/reference/api).
+
+## Configuring Tracing in Dapr
+As mentioned earlier, in this tracing example, we are using the same [grpc app](../create-grpc-app).
```
Comment (orizohar) [inline howto/use-w3c-tracecontext/README.md:144]: Please rephrase to remove all first person references ("we", "I")

File: howto/use-w3c-tracecontext/README.md [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+
+To view traces, you need to deploy OpenCensus supported exporters. This is independent of Dapr configuration.
+You can refer [how-to-diagnose-with-tracing](../diagnose-with-tracing) to set up trace exporters. 
+
+## Invoking Dapr With Trace Context
+
+As mentioned earlier in the [section](#Using-Trace-Context-in-Dapr), you can create the trace context and pass it through when calling Dapr or Dapr can generate trace context passed it back to you. 
+
+If you choose to pass the trace context explictly, then Dapr will use the passed trace context and propagate all across the HTTP/gRPC call.
+We are using the same [grpc app](../create-grpc-app), however listing out the complete code for putting it all together.
```
Comment (orizohar) [inline howto/use-w3c-tracecontext/README.md:181]: Please rephrase to remove all first person references ("we", "I")

File: howto/use-w3c-tracecontext/README.md [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+			Method:      "MyMethod",
+			ContentType: "text/plain; charset=UTF-8",
+			Data:        &any.Any{Value: []byte("Hello")},
+		},
+	})
+```
+
+That's it !. Now you can correlate the calls in your app and in Dapr across services using the same trace context.
+
+To view traces, you can refer [how-to-diagnose-with-tracing](../diagnose-with-tracing) e.g Zipkin/Application Insights.
```
Comment (orizohar) [inline howto/use-w3c-tracecontext/README.md:242]: This should be a "related links" section so it's more readable. Please add bullets for related links:
- general "observability" 
- observability sample
- Zipkin how-to
- repeat the W3C tracing context spec link here so it is easy to find again

File: howto/use-w3c-tracecontext/README.md [hunk 8]
```diff
@@ -0,0 +1,243 @@
...
+
+Traces that are collected by different tracing vendors cannot be correlated as there is no shared unique identifier.
+Traces that cross boundaries between different tracing vendors can not be propagated as there is no uniformly agreed set of identification that is forwarded.
+Vendor specific metadata might be dropped by intermediaries.
+Cloud platform vendors, intermediaries and service providers, cannot guarantee to support trace context propagation as there is no standard to follow.
+In the past, these problems did not have a significant impact as most applications were monitored by a single tracing vendor and stayed within the boundaries of a single platform provider. Today, an increasing number of applications are highly distributed and leverage multiple middleware services and cloud platforms.
+
+This transformation of modern applications calls for a distributed tracing context propagation standard.
+
+W3C trace context specification defines a universally agreed-upon format for the exchange of trace context propagation data - referred to as trace context. Trace context solves the problems described above by
```
Comment (orizohar) [inline howto/use-w3c-tracecontext/README.md:13]: The link to the context (which is provided further down) might be better placed on the words "W3C trace context" here

File: howto/use-w3c-tracecontext/README.md [hunk 9]
```diff
@@ -0,0 +1,243 @@
+# W3C Trace Context - Background
+
+Distributed tracing is a methodology implemented by tracing tools to follow, analyze and debug a transaction across multiple software components. Typically, a distributed trace traverses more than one component which requires it to be uniquely identifiable across all participating systems. Trace context propagation passes along this unique identification. Today, trace context propagation is implemented individually by each tracing vendor. In multi-vendor environments, this causes interoperability problems, like:
```
Comment (orizohar) [inline howto/use-w3c-tracecontext/README.md:3]: This section seems to explain what a trace context is. This is great but the sub section should be named accordingly.

Comment (orizohar) [inline howto/use-w3c-tracecontext/README.md:3]: This article has several sections, please add a "contents" section with links to the sub headers for easy navigation

File: howto/use-w3c-tracecontext/README.md (hunk 1 above)
Comment (shalabhms) [reply howto/use-w3c-tracecontext/README.md]: done.

File: howto/use-w3c-tracecontext/README.md (hunk 2 above)
Comment (shalabhms) [reply howto/use-w3c-tracecontext/README.md]: done.

File: howto/use-w3c-tracecontext/README.md (hunk 9 above)
Comment (shalabhms) [reply howto/use-w3c-tracecontext/README.md:9]: done.

Comment (shalabhms) [reply howto/use-w3c-tracecontext/README.md:9]: done.

**Comment by shalabhms** [comment]
> @shalabhms - this is very well written! On top of my other comments, please add to this PR changes to other docs that will link into this one (in their respective related links sections) such as:
> 
> * Main How To README - https://github.com/dapr/docs/tree/master/howto#observerability
> * Observability sample
> * Zipkin How To
> * Azure Monitor tracing How To
> * Observability  concept https://github.com/dapr/docs/blob/master/concepts/observability/README.md
> * Tracing concept https://github.com/dapr/docs/blob/master/concepts/observability/traces.md


Thanks @orizohar :-) . I have addressed the comments.

**Comment by orizohar** [comment]
Thanks for addressing the comments. Just missing a link from main HowTo README https://github.com/dapr/docs/tree/master/howto#observerability

**Pull request merged**